vertx-template-engines/vertx-web-templ-rocker/src/main/java/io/vertx/ext/web/templ/impl/ArrayOfByteBufsOutput.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.ext.web.templ.impl;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fizzed.rocker.ContentType;
+import com.fizzed.rocker.runtime.AbstractRockerOutput;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.buffer.UnpooledByteBufAllocator;
+
+/**
+ * @author <a href="mailto:xianguang.zhou@outlook.com">Xianguang Zhou</a>
+ */
+public class ArrayOfByteBufsOutput extends AbstractRockerOutput<ArrayOfByteBufsOutput> {
+
+  private final List<ByteBuf> arrays;
```

Comment:
I think we should use only vertx 'Buffer' to append 'byte[]' or 'String' with encoding

Reply:
Using Vertx "Buffer" to append "byte[]" or "String" with encoding will increase unnecessary memory copies.

Reply:
@Xianguang-Zhou I have to agree with @EmadAlblueshi here, vert.x Buffer is just a wrapper around netty `ByteBuf` so it safe guards us from netty API changes and reduces boilerplate.

For most cases I don't see how using netty directly will have a severe performance impact, so I am of the opinion that we could first keeps things simple using Buffer and in there is a real need maybe have a low level netty alternative configurable from the engine.